### PR TITLE
DEVPROD-36629: Reuse project across patch scheduling

### PR DESCRIPTION
**Pull request by sophstad**
DEVPROD-36629 - tagging @malikchaya2 because of your caching work in DEVPROD-36139! I don't think this conflicts with the project (or is covered by another ticket) but wanted to get your thoughts :)

### Description

`FindAndTranslateProjectForPatch` is called 2-3 times per schedule operation in order to obtain the translated project from the config. We can pass it between functions quite easily instead, since these operations are sequential.

### Testing

- **Before:** [staging trace](https://ui.honeycomb.io/mongodb-4b/environments/staging/datasets/evergreen/result/nHogVDEksZc/trace/xY5udwTAKTc?fields[]=s_name&fields[]=s_serviceName&span=e6f6c1bfb4f70424) where `finalize-patch` has 19 child spans
- **After:** [staging trace](https://ui.honeycomb.io/mongodb-4b/environments/staging/datasets/evergreen/result/uyGJVURgg7h/trace/u4d792BGxLt?fields[]=s_name&fields[]=s_serviceName&span=7536edcc55b9bde5) where `finalize-patch` has 13 child spans

### CODE REVIEW

File: graphql/mutation_resolver.go [hunk 1]
```diff
@@ -424,7 +424,7 @@ func (r *mutationResolver) SchedulePatch(ctx context.Context, patchID string, co
 	if err != nil && !adb.ResultsNotFound(err) {
```
Comment (malikchaya2) [inline graphql/mutation_resolver.go:424]: (random comment placement) 

This path is not cached because the translation cache only pays off for configs translated repeatedly across independent requests, whereas an unfinalized patch is a short-lived, single-flow object. 

This change is great! It eliminates the waste on that path of re-translating the same patch three times in one request, which is something caching structurally couldn't fix.

File: model/patch_lifecycle_test.go [hunk 2]
```diff
@@ -306,7 +306,7 @@ func TestFinalizePatch(t *testing.T) {
 			p.ProjectStorageMethod = ppStorageMethod
```
Comment (malikchaya2) [inline model/patch_lifecycle_test.go:306]: Can we add a test that asserts that passing a non-nil translatedProject yields the same version/tasks as re-translating from the DB?

**Review by malikchaya2 — APPROVED** [review]
Nice work!
